c7n/resources/s3.py

```diff
@@ -1416,35 +1417,26 @@ def process(self, buckets, event=None):
 
     def process_bucket(self, bucket):
         s3 = bucket_client(local_session(self.manager.session_factory), bucket)
-        enabled = self.data.get('enabled', False)
-        scope = self.data.get('scope', 'Any')
+        config = {}
         try:
             config = s3.get_public_access_block(
                 Bucket=bucket['Name'])['PublicAccessBlockConfiguration']
         except ClientError as e:
             if e.response['Error']['Code'] == 'NoSuchPublicAccessBlockConfiguration':
-                # Set config to none because NoSuchPublicAccessBlockConfiguration error
-                # was returned. This is important later because it is symbolic of
-                # all blocks being set to false. See if/else in matches_filter func
-                config = None
+                # When there is a None config, set all to false to be symbolic
+                for key in self.keys:
+                    config[key] = False
             else:
                 raise
-        if self.matches_filter(config, enabled, scope):
-            return {"Name": bucket['Name'], "publicblocks": config}
-
-    def matches_filter(self, config, enabled, scope):
-        if config:
-            if scope == 'All':
-                return all(config.values()) if enabled else not any(config.values())
-            elif scope == 'Any':
-                return any(config.values()) if enabled else not all(config.values())
-            else:
-                return config[scope] if enabled else not config[scope]
+        if self.matches_filter(config):
+            return {"Name": bucket['Name'], "PublicAccessBlockConfiguration": config}
```

Comment:
for #5567, shouldn't this be modifying `bucket` to add the `PublicAccessBlockConfiguration` to it and then returning `bucket` itself?

Reply:
And if we do that, we should probably be checking if `PublicAccessBlockConfiguration` is already in `bucket` before spinning up the s3 client and attempting the api call... so if someone chains these filters, it just reuses the config from the first one.

Reply:
This was an oversight when playing with how to minimize response data. I had not considered how that stacks with other filters. Will address on the next commit.

Reply:
That’s a good call, it could save some call outs.

Reply:
Resolved in e2f88f3. Thanks for recommending the cache check!